src/geometry_aux.cpp

```diff
@@ -395,19 +414,30 @@ prepare_distribcell()
 void
 count_cell_instances(int32_t univ_indx)
 {
-  for (int32_t cell_indx : model::universes[univ_indx]->cells_) {
-    Cell& c = *model::cells[cell_indx];
-    ++c.n_instances_;
 
-    if (c.type_ == FILL_UNIVERSE) {
-      // This cell contains another universe.  Recurse into that universe.
-      count_cell_instances(c.fill_);
+  if (model::universe_cell_counts.count(univ_indx)) {
+    std::map<int32_t, int> univ_counts = model::universe_cell_counts[univ_indx];
```

Comment:
```suggestion
    const auto& univ_counts = model::universe_cell_counts[univ_indx];
```
Also, calling `count` above and then indexing the map will result performing the same (O(log N)) search twice. It would be better to use `find` to get an iterator to the element and check if it is `== end()` so that the search is only done once.